### PR TITLE
ceph: add physical device labels to osd pods

### PR DESCRIPTION
**Pull request by mvollman**
Signed-off-by: Michael Vollman <michael.b.vollman@gmail.com>

**Description of your changes:**
Add a ceph-osd-device label to each OSD pod and deployment with a value
of the physical device path of the drive associated with the OSD.
Due to labeling constraints the leading forward slash will be dropped
from the path and all other forward slashes are converted to underscores
so /dev/sdb label value wiill be dev_sdb

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [X] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [X] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](https://github.com/rook/rook/blob/master/CONTRIBUTING.md#comments)
- [ ] Add the flag for skipping the CI if this PR does not require a build. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for more details.

### CODE REVIEW

**Comment by mykaul** [comment]
It'd be great if we could use a more stable device name (/dev/disk/by-uuid based for example). Do we have this information anywhere?

File: pkg/daemon/ceph/osd/volume.go [hunk 1]
```diff
@@ -220,6 +222,8 @@ func getCephVolumeOSDs(context *clusterd.Context, clusterName string, cephfsid s
 			if osd.Type == "journal" {
 				isFilestore = true
 			}
+			osdDevice = strings.Replace(osd.Devices[0], "/", "", 1)
+			osdDevice = strings.Replace(osdDevice, "/", "_", -1)
```
Comment (leseb) [inline pkg/daemon/ceph/osd/volume.go]: Can't we just get `sdb` instead of `dev_sdb`, I don't see much value in having `dev_`.

**Comment by leseb** [comment]
> It'd be great if we could use a more stable device name (/dev/disk/by-uuid based for example). Do we have this information anywhere?

I second that but at this point, I don't even think there is a use case remembering which drive the OSD is running on. I don't think we have the `/dev/disk/by-` info handy, so if the machine reboots and drives letter change this info won't mean anything anymore.

I believe using the OSD FSIF as an attribute to find the LV consumed by that OSD is better. You can find this pretty easily by calling `ceph-volume lvm list`.

**Comment by mvollman** [comment]
Yea the `/dev/disk/by-` information is not readily available.  Just the information returned from `ceph-volume lvm list` is available for tracking.  I was just looking for a way to track physical devices without having to enter a container and run `ceph-volume lvm list` manually.  I believe you are both right that if a drive letter changes the `ceph-osd-device` value will not mean anything anymore.  I am going to mock this scenario in my lab just to confirm the behavior.

File: pkg/daemon/ceph/osd/volume.go (hunk 1 above)
Comment (mvollman) [reply pkg/daemon/ceph/osd/volume.go]: fine with me if there is a desire to move forward with this change.

**Comment by mvollman** [comment]
I was able to confirm that if an OSD is reseated such that its drive letter changes, the OSD pod is able to recover operations using the new drive letter while the label still reflects the original drive letter.  I will go ahead and close this pull request.

**Pull request closed without merging**